main.cpp

```diff
@@ -115,18 +121,42 @@ int main()
     /*
      construct 2 more Car instances via emplace_back.
      */
-    
+    cars.emplace_back("toby");
+    cars.emplace_back("kyle");
+
     /*
      now reserve and emplace_back your Trucks and Motorcyles
      */
-    
-    
-    
-    
-    assert(false);
+    motorcycles.reserve(2);
+    motorcycles.emplace_back("jeff");
+    motorcycles.emplace_back("yo");
+
+    trucks.reserve(6);
+    trucks.emplace_back("sarah");
+    trucks.emplace_back("bob");
+    trucks.emplace_back("jess");
+    trucks.emplace_back("jojo");
+    trucks.emplace_back("jack");
+    trucks.emplace_back("ken");
+
     //add the cars, motorcycles and trucks to the highway using range-based for() loops: for( element : vec ) { ... }
     //be careful to not accidentally make element copies when iterating.
-    
+
+    for( auto& car : cars)
+    {
+        highway.addVehicle(&car);
+    }
+
+    for( Motorcycle& motorcycle : motorcycles)
+    {
+        highway.addVehicle(&motorcycle);
+    }
+
+    for ( SemiTruck& truck : trucks)
+    {
+        highway.addVehicle(&truck);
+    }
```

Comment:
use `auto&` in case you decide to change the type stored in the container
- [ ] revise